pkg/deploy/plugins/namespace.go

```diff
@@ -0,0 +1,42 @@
+package plugins
+
+import (
+	"errors"
+	"fmt"
+
+	"sigs.k8s.io/kustomize/api/resmap"
+)
+
+// CreateNamespacePlugin creates a new namespace plugin.
+func CreateNamespacePlugin(namespace string) *namespacePlugin {
+	return &namespacePlugin{
+		namespace: namespace,
+	}
+}
+
+type namespacePlugin struct {
+	namespace string
+}
+
+// Config implements the TransformerPlugin interface.
+func (p *namespacePlugin) Config(h *resmap.PluginHelpers, config []byte) error {
+	return nil
+}
+
+// Transform implements the TransformerPlugin interface.
+func (p *namespacePlugin) Transform(m resmap.ResMap) error {
+	// do not transform an invalid namespace
+	if p.namespace == "" {
+		return errors.New("failed to set namespace: namespace cannot be empty")
+	}
+	for _, res := range m.Resources() {
+		// skip cluster-scoped resources because they don't have a namespace
+		if res.GetGvk().IsClusterScoped() {
+			continue
+		}
+		if err := res.SetNamespace(p.namespace); err != nil {
```

Comment:
So if I understand right, it will override the namespace unconditionally even when its already set on the resource, and not only when the namespace is missing ?

Reply:
@VaishnaviHire what do you think? I'm okay with the namespace plugin as a forced write, but is that intended behavior?

Reply:
@rhuss Since these are internal manifests, managed by the operator explicitly, we do want to override any `namespace` value that doesn't correspond to the given namespace. We are creating these runtime plugins to override the defaults in the manifests.